backend/app/core/mail.py

```
@@ -25,88 +27,83 @@ async def get_user_email(user_id: int, db: AsyncSession) -> Optional[str]:
-        to_email=email,
-        subject=subject,
-        body=body
-    )
-    await send_email(email_data)
+            with smtplib.SMTP(settings.MAIL_SERVER, settings.MAIL_PORT) as server:
+                if settings.MAIL_STARTTLS:
+                    server.starttls()
+                server.login(settings.MAIL_USERNAME, settings.MAIL_PASSWORD)
+                server.send_message(msg)
+        except Exception as e:
+            raise HTTPException(status_code=500, detail=f"Failed to send email: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)